Description: Use VPN plugin paths as specified in the .name files
 Since LIBDIR of gnome-control-center and the VPN plugins aren't necessarily
 the same, use the paths as specified by the VPN plugins and only
 reconstruct the path if it is not absolute or we fail to load the plugin
Author: Michael Biebl <biebl@debian.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=743333
Index: gnome-control-center-3.8.3/panels/network/connection-editor/vpn-helpers.c
===================================================================
--- gnome-control-center-3.8.3.orig/panels/network/connection-editor/vpn-helpers.c	2014-04-03 00:45:24.297705388 +0200
+++ gnome-control-center-3.8.3/panels/network/connection-editor/vpn-helpers.c	2014-04-03 00:45:48.994026328 +0200
@@ -75,7 +75,7 @@
 		char *path = NULL, *service = NULL;
 		char *so_path = NULL, *so_name = NULL;
 		GKeyFile *keyfile = NULL;
-		GModule *module;
+		GModule *module = NULL;
 		NMVpnPluginUiFactory factory = NULL;
 
 		if (!g_str_has_suffix (f, ".name"))
@@ -95,19 +95,24 @@
 		if (!so_path)
 			goto next;
 
-		/* Remove any path and extension components, then reconstruct path
-		 * to the SO in LIBDIR
-		 */
-		so_name = g_path_get_basename (so_path);
-		g_free (so_path);
-		so_path = g_build_filename (NM_VPN_MODULE_DIR, so_name, NULL);
-		g_free (so_name);
+		if (g_path_is_absolute (so_path))
+			module = g_module_open (so_path, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
 
-		module = g_module_open (so_path, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
 		if (!module) {
-			g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED, "Cannot load the VPN plugin which provides the "
-			             "service '%s'.", service);
-			goto next;
+			/* Remove any path and extension components, then reconstruct path
+			 * to the SO in LIBDIR
+			 */
+			so_name = g_path_get_basename (so_path);
+			g_free (so_path);
+			so_path = g_build_filename (NM_VPN_MODULE_DIR, so_name, NULL);
+			g_free (so_name);
+
+			module = g_module_open (so_path, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
+			if (!module) {
+				g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED, "Cannot load the VPN plugin which provides the "
+				             "service '%s'.", service);
+				goto next;
+			}
 		}
 
 		if (g_module_symbol (module, "nm_vpn_plugin_ui_factory", (gpointer) &factory)) {
